### ISSUE TITLE
Request that compilation with -WEverything should compile cleanly

### ISSUE THREAD

**Issue by drpatrik** [eidheim/Simple-Web-Server#44]
Really great work however in our code base we always compile with -WEverything and -Werror under clang which produce a lot of warnings hence errors. I am willing to fix all the warning and issue a pull request as long as you are fine with it?

You have the habit of using the same name for parameter and member variable which produce a warning such as "declaration shadows a local variable". To fix that I need to introduce a naming convention such as member_ or something similar and I would like to know, in that case, what you prefer?

**Comment by jeffreyscottgraham** [comment]
I second that!

**Comment by eidheim** [comment]
Sure, what constructor parameter name convention is most common?

Also, with regards to this, and since I've always wondered: is having unused parameters in functions a slight performance issue compared to commenting them out if not used?

**Comment by drpatrik** [comment]
We are using google code style and they use "foo_", other uses "_foo" and if you are used to the Microsoft world "m_foo" is also used. I prefer "foo_" but its your code base so you should decided.

I would say that I believe that its not very easy for a compiler to ignore building the full parameter list even if there is a type but no parameter name (e.g. void foo(int j, int /*i*/) {}). If you have a function void foo(int /*i*/, int j) the compiler must reserve space on the stack for i since j is used (in this case however the compiler can choose to pass parameters in registers instead of using stack space). Its fairly easy to verify however by looking at the assembler generated.

That said I think that you can safety ignore the overhead of parameter passing as long as the profiler does not prove me wrong by indicates that its a performance bottleneck.

**Comment by eidheim** [comment]
Thank you again. I'm fine with `foo_`.

**Comment by drpatrik** [comment]
I did some small tests since I found your question quite interesting. What I found was that in debug the parameter was always pushed and when I compiled with full optimisation the function was inlined. If we want to test better we must use more realistic function sizes but then the cost will be much harder to read asm-ouput :-)

I will fix all the warnings and provide a pull-request probably tomorrow. Thanks for all your work!
